### PR TITLE
[CI] Stabilize multinode DP internal LB completion tests

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
Fixes flaky `test_api_only_multinode_dp_completion` and `test_multinode_dp_completion` which intermittently fail with empty model responses during concurrent load balancer testing.

### Motivation

```
> assert len(choice.text) >= 1
E AssertionError: assert 0 >= 1
E  +  where 0 = len('')
```

These tests intentionally use `temperature=1.0` to produce diverse outputs across 200 concurrent requests for realistic load balancer distribution testing. However, at `temperature=1.0` the model can legitimately emit a stop token as its very first token, producing `text=''` with `finish_reason='stop'`. Over 400 requests (two bursts of 200), the probability of at least one empty response is high. Rather than changing `temperature=0.0` (which would undermine the test's intent of exercising load balancing with diverse requests), the fix tolerates the valid edge case: when `finish_reason='stop'`, empty text is accepted. The non-empty text assertion is only enforced when `finish_reason='length'`.

- **`_make_completion_request` helper**: Extracted the duplicated `make_request()` closure from both non-streaming completion tests into a shared module-level function with diagnostic assertion messages that print actual values on failure.

- **`_run_request_bursts` helper**: Extracted the duplicated two-burst loop pattern (create tasks -> gather -> validate -> sleep) shared by both non-streaming tests.

- **Streaming tests unchanged**: They already use `temperature=0.0` and have adequate assertions.

cc @kenroche

### CODE REVIEW

File: tests/v1/distributed/test_internal_lb_dp.py [hunk 1]
```diff
@@ -27,6 +27,79 @@
...
+            )
+            await asyncio.sleep(0.01)
+
+        results = await asyncio.gather(*all_tasks)
+        assert len(results) == num_requests, (
+            f"Burst {burst}: expected {num_requests} results, got {len(results)}"
+        )
+        assert all(completion is not None for completion in results), (
+            f"Burst {burst}: some completions were None"
+        )
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply tests/v1/distributed/test_internal_lb_dp.py]: Using return_exceptions=True and re-raising to ensure clean task shutdown. Done :)

**Review by njhill — APPROVED** [review]
Thanks @AndreasKaratzas
